pywebostv/controls.py

```diff
@@ -52,6 +55,17 @@ def standard_validation(payload):
     return True, None
 
 
+def get_icon_data(url: str):
+    """Return base64 encoded image data retrieved from url."""
+    if os.path.exists(url):
+        with open(url, 'rb') as f:
+            content = f.read()
+    else:
+        content = requests.get(url, timeout=17).content
```

Comment:
I think the library shouldn't be making HTTP GET requests to retrieve the image. It should be left to the caller of this library. On the same note, it shouldn't be assuming it can directly read from the filesystem too. It should be the callers responsibility. The caller should just pass the appropriate bytes and extension to this API.

Like:
 
```python
content = requests.get(...).content
system.notify("message", icon_bytes=content, icon_ext="png")
```

Can you also please remove the `import requests` & `import os` lines above?

Reply:
Makes sense, a lot cleaner now.